### ISSUE TITLE
Godot fails to compile on mingw

### ISSUE THREAD

**Issue by fire** [godotengine/godot#13949]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
WSL on Fedora compiling for Windows for MINGW64

**Issue description:**
```
Linking Program        ==> bin/godot.windows.opt.tools.64.exe
modules/theora/video_stream_theora.cpp: In member function 'set_file':
thirdparty/libtheora/decode.c:400:49: warning: iteration 2 invokes undefined behavior [-Waggressive-loop-optimizations]
       qsum+=_dec->state.dequant_tables[qti][pli][qi][12]+
                                                 ^
thirdparty/libtheora/decode.c:396:3: note: within this loop
   for(qi=0;qi<64;qi++){
   ^
thirdparty/zstd/compress/zstd_compress.c: In function 'ZSTD_getCParams':
thirdparty/zstd/compress/zstd_compress.c:3012:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 }
 ^
thirdparty/zstd/compress/zstd_compress.c: In function 'ZSTD_freeCCtx':
thirdparty/zstd/compress/zstd_compress.c:118:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 }
 ^
lto1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
thirdparty/libwebp/enc/picture_csp_enc.c: In function 'WebPPictureHasTransparency':
thirdparty/libwebp/enc/picture_csp_enc.c:70:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 }
 ^
thirdparty/libwebp/enc/config_enc.c: In function 'WebPValidateConfig':
thirdparty/libwebp/enc/config_enc.c:129:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 }
 ^
thirdparty/libwebp/utils/utils.c: In function 'WebPSafeMalloc':
thirdparty/libwebp/utils/utils.c:198:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 }
 ^
thirdparty/libwebp/enc/picture_enc.c: In function 'WebPMemoryWrite':
thirdparty/libwebp/enc/picture_enc.c:217:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 }
 ^
thirdparty/libwebp/enc/picture_csp_enc.c: In function 'WebPPictureImportRGBA':
thirdparty/libwebp/enc/picture_csp_enc.c:1152:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 }
 ^
thirdparty/libwebp/enc/picture_csp_enc.c: In function 'WebPPictureImportRGB':
thirdparty/libwebp/enc/picture_csp_enc.c:1138:1: warning: visibility attribute not supported in this configuration; ignored [-Wattributes]
 }
 ^
/tmp/ccaz4kc3.ltrans0.ltrans.o:<artificial>:(.text+0xd33da): undefined reference to `__imp_AddDllDirectory'
/tmp/ccaz4kc3.ltrans0.ltrans.o:<artificial>:(.text+0xd340f): undefined reference to `__imp_RemoveDllDirectory'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.windows.opt.tools.64.exe] Error 1
scons: building terminated because of errors.
```

**Steps to reproduce:**

Compile godot on mingw64 and fails to compile. 

**Steps to fix?**

Remove the RemoveDllDirectory and AddDllDirectory usage.

See: https://github.com/godotengine/godot/commit/9678231b109c333a5273325c8758241310cd27f4

**Comment by fire** [comment]
Is also a windows 8 only api..

**Comment by fire** [comment]
https://github.com/webrtc-uwp/chromium-base/blob/master/native_library_win.cc#L48 has interesting ways of checking for the availability of functions.

**Comment by bruvzg** [comment]
Add(Remove)DllDirectory functions should be available on Windows 7 with [KB2533623](https://support.microsoft.com/en-us/help/2533623/) patch

**Comment by maxim-sheronov** [comment]
Also this happens if build on Linux machine with mingw-w64 with target=windows

**Comment by Xrayez** [comment]
Same, using similar configuration, WSL aswell but on Ubuntu.

**Comment by maxim-sheronov** [comment]
Checked on Ubuntu 16.04 with mingw 4.0 and Ubuntu 18.04 with latest mingw 5.0.3. Same problem.

**Comment by maxim-sheronov** [comment]
@akien-mga Why did you remove build check with GODOT_TARGET=windows from travis builds? It could prevent this bug.

**Comment by drentsoft** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  drentsoft@gmail.com
    Domain drentsoft.com has exceeded the max emails per hour (506/500 (101%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext4.iad.github.net ([192.30.252.195]:52138 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by wdc002.hawkhost.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.89)
	(envelope-from <noreply@github.com>)
	id 1eMTQB-000kjf-OB
	for info@drentsoft.com; Wed, 06 Dec 2017 01:43:07 -0500
Date: Tue, 05 Dec 2017 22:43:07 -0800
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1512542587;
	bh=sjSOfMpfgWm8WwgF26z+fE2Y21IYibJUCCspAplPFGg=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=H5NtMi1wEDUpm2mDQHOvKRaUR2k7zrrm93ZkAqN7bTHi35M0yxZj6WGcu5HiZPlD9
	 ynjNCUpqN9gLiGE7XOAFil+kNeNfjNxik9iQ2j+aE3TNJh+Ij7fAaqQ6h0YbmWKpJ+
	 V5H6g4ADvbIKpF5YskR/6Dulfip3Lqash97/cdgc=

**Comment by akien-mga** [comment]
Fixed by #14313.
